sklearn/metrics/pairwise.py

```diff
@@ -1985,16 +2002,29 @@ def _pairwise_callable(X, Y, metric, ensure_all_finite=True, **kwds):
         # No input dimension checking done for custom metrics (left to user)
         ensure_2d=False,
     )
+    _, _, dtype_float = _find_floating_dtype_allow_sparse(X, Y, xp=xp)
+
+    def _get_slice(array, index):
+        # TODO : Support for 1D shapes in scipy sparse arrays (COO, DOK and CSR
+        # formats) only added in 1.14. We must return 2D array until min scipy 1.14
+        # i.e. below 2 lines can be removed once min scipy >= 1.14
```

Comment:
Let's make this TODO more likely to match maintainers grep commands, with both "scipy" and "1.14" on the same line as the TODO marker:

```suggestion
        # TODO: simplify once we drop support for scipy < 1.14. Support for
        # 1D shapes in scipy sparse arrays (COO, DOK and CSR formats) only
        # added in 1.14. We must return 2D array until min scipy 1.14 i.e.
        # below 2 lines can be removed once min scipy >= 1.14
```